webpack/webpack.common.cjs

```diff
@@ -82,7 +82,8 @@ const config = {
     }),
     new webpack.DefinePlugin({
       VERSION: JSON.stringify(require("../package.json").version),
-      "process.env": JSON.stringify(process.env),
+      "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV),
```

Comment:
this is bc we define a new plugin in webpack.common.cjs:
```
new webpack.DefinePlugin({
      VERSION: JSON.stringify(require("../package.json").version),
      "process.env.NODE_ENV": JSON.stringify(process.env.NODE_ENV),
      "process.env": JSON.stringify(process.env)
    })
```

In order to use `process.env.NODE_ENV` in `useActivityTimer.tsx`
`const timeout = process.env.NODE_ENV === "development" ? 3600000 : 60000; // 1h/1min`

Reply:
I tested this and it seems to fix it, can you please check @jimcase :

![07052024113806](https://github.com/cardano-foundation/cf-identity-wallet/assets/119612231/d92aa24b-aa35-46d7-91de-c3996372056f)

Reply:
Wouldn't that make it always development mode?

Reply:
Sorry, can try this:

```
  mode: new URLSearchParams(window.location.search).has("browserPreview")
    ? "development"
    : "production",
```

Reply:
That would be development mode on Vercel and production on local. If you meant the other way around, that still isn't the goal because our builds deployed on mobile would be development.

The point is development mode is when hot reloading with the Webpack server, and production is everything else.

Reply:
The warning is gone but in this way we hardcode the value of `process.env.NODE_ENV`, so it is always `"development"` and never `production`.

Instead, i defined this plugin in `webpack.dev.cjs` instead of `webpack.common.cjs`, so its not hardcoded in the common config file.

Reply:
Sorry, that was just to fix the warning, I meant to add this as well:

```
import { getPlatforms } from "@ionic/react";
const platforms = getPlatforms();

...

mode: platforms.includes("mobileweb") ? "development" : "production",

...

    new webpack.DefinePlugin({
      VERSION: JSON.stringify(require("../package.json").version),
      "process.env.NODE_ENV": JSON.stringify("development"),
      "process.env": JSON.stringify(process.env),
    }),
```

Reply:
Except vercel should be a production version. Change pushed by Jaime now looks good to me now though

Reply:
Yup, happy with that 👍